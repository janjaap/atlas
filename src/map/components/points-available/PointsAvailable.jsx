import React from 'react';
import PropTypes from 'prop-types';

import './_points-available.scss';

const PointsAvailable = ({ markersLeft }) => {
  const label = markersLeft === 0 ?
    'Geen punten meer mogelijk' :
    `Nog ${markersLeft} punt${markersLeft !== 1 ? 'en' : ''} mogelijk`;

  return (
    <div className="points-available">
      <span
        aria-label={label}
        className="points-available__label"
      >{label}</span>
    </div>
  );
};

PointsAvailable.propTypes = {
  markersLeft: PropTypes.number.isRequired
};

export default PointsAvailable;
