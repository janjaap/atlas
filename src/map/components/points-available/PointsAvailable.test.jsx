import React from 'react';
import { shallow } from 'enzyme';

import PointsAvailable from './PointsAvailable';

describe('PointsAvailable', () => {
  it('should render no markers left', () => {
    const wrapper = shallow(
      <PointsAvailable
        markersLeft={0}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one marker left', () => {
    const wrapper = shallow(
      <PointsAvailable
        markersLeft={1}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render multiple markers left', () => {
    const wrapper = shallow(
      <PointsAvailable
        markersLeft={2}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
