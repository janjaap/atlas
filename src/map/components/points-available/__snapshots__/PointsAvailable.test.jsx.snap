// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsAvailable should render multiple markers left 1`] = `
<div
  className="points-available"
>
  <span
    aria-label="Nog 2 punten mogelijk"
    className="points-available__label"
  >
    Nog 2 punten mogelijk
  </span>
</div>
`;

exports[`PointsAvailable should render no markers left 1`] = `
<div
  className="points-available"
>
  <span
    aria-label="Geen punten meer mogelijk"
    className="points-available__label"
  >
    Geen punten meer mogelijk
  </span>
</div>
`;

exports[`PointsAvailable should render one marker left 1`] = `
<div
  className="points-available"
>
  <span
    aria-label="Nog 1 punt mogelijk"
    className="points-available__label"
  >
    Nog 1 punt mogelijk
  </span>
</div>
`;
